source/PCA9685Support.cpp

```diff
@@ -400,32 +415,13 @@ HRESULT PCA9685Device::_InitializeChip(ULONG i2cAdr)
             transaction.useHighSpeed();
 
             // Queue sending the address of the MODE1 regeister to the PWM chip.
-            hr = transaction.queueWrite(mode1RegAdr, sizeof(mode1RegAdr));
+            hr = transaction.queueWrite(mode1RegAdr, sizeof(mode1RegAdr), TRUE);
         }
 
         if (SUCCEEDED(hr))
         {
-            // Queue reading the contents of the MODE1 register.
-            hr = transaction.queueRead(readBuf, sizeof(readBuf));
-        }
-
-        //
-        // If the SLEEP bit is clear, the chip has been initialize: abort the rest of the transaction.
-        //
-
-        if (SUCCEEDED(hr))
-        {
-            // Register a callback to test if the chip is initialized.
-            hr = transaction.queueCallback([&readBuf, &transaction]()
-            {
-                PMODE1 mode1RegPtr = (PMODE1)readBuf;
-                if (mode1RegPtr->SLEEP == 0)   // If chip is not in sleep mode,
-                {
-                    transaction.abort();
-                }
-                return S_OK;
-            }
-            );
+            // Put the chip to sleep.
+            hr = transaction.queueWrite((PUCHAR)&mode1RegSleep, 1);
```

Comment:
We should discuss offline what problems you encountered and how they should be fixed.  There should not be a need to put the chip to sleep to reset it or get it back to a known state.  If you do put it to sleep, you need to properly handle side-effects that are caused by this operation (oscillator restart, use of the chip by other processes).